File: problemasOlimpicos/danger.cpp
#include <cstdio>
#include <iostream>
#include <string>

using namespace std;
int main()
{
	std::string s;
	std::cin >> s;
	int seq = 1;
	int j = 1;
	for(int k = 1; k <= s.length(); k++)
	{
		if(s[k] == s[k - 1])
		{
			seq++;
			if(seq == 7)
				j++;
		}
		else
			seq = 1;
	}
	if(j > 1)
		printf("YES\n");
	else
		printf("NO\n");
	return 0;
}

File: problemasOlimpicos/danger.py
x = raw_input()
seq = 1
i = 1
a = 0
while(i < len(x)):
	if(x[i] == x[i - 1]):
		seq = seq + 1
		i = i + 1
		if(seq == 7):
			a = a + 1
	else:
		seq = 1
		i = i + 1

if(a > 0):
	print("YES")
else:
	print("NO")